mne/viz/tests/test_raw.py

```diff
@@ -62,26 +36,27 @@ def _annotation_helper(raw, events=False):
     fig = raw.plot(events=events)
     assert len(plt.get_fignums()) == 1
     data_ax = fig.mne.ax_main
-    fig.canvas.key_press_event('a')  # annotation mode
-    assert len(plt.get_fignums()) == 2
+    fig.press_key('a')  # annotation mode
+    assert fig.get_n_windows() == 2
     # +3 from the scale bars
     n_scale = 3
     assert len(data_ax.texts) == n_anns + n_events + n_scale
     # modify description to create label "BAD test"
     ann_fig = fig.mne.fig_annotation
     for key in ['backspace'] + list(' test;'):  # semicolon is ignored
-        ann_fig.canvas.key_press_event(key)
+        fig.press_key(key, target=ann_fig)
```

Comment:
I don't love this (would have expected `ann_fig.press_key(key)`).  I guess it's a consequence of the fact that pyqtgraph backend will handle annotations through its toolbar and hence won't have a separate annotation window?  Would it work to have `ann_fig` still exist in pyqtgraph case, but have it be a non-displaying object that passes interactions back to the main window?  Not sure which way is easier / better; WDYT?

Reply:
Frankly applying additional structures from matplotlib onto pyqtgraph (and then probably also vice versa) seems to me a bit confusing and constraining regarding the further development of each backend. 
But if we want minimal change and we don't want to have if-cases for each backend in some of the tests, there probably has to be a dummy-counterpart or something similar. This might be more conservative on the tests if these cases are handled inside the respective Backend-Class (with target/fig as parameter and not as potential non-existing object).

What may be a solution to this specific problem could be putting this method and also e.g _fake_click into MNEFigure. 
But this would then make inheritance and having those methods as abstract methods in BrowserBase more difficult (wasn't able to solve that yet, so I first await for your response about this).
WDYT?

Reply:
> But if we want minimal change and we don't want to have if-cases for each backend in some of the tests, there probably has to be a dummy-counterpart or something similar.

This is certainly my preference (compared to using if-else logic for the different backends).  Having an explicit dummy method provides a place in the code to briefly explain *why* that dummy method is a dummy (e.g., a docstring that says "dummy method for X, because PyQtGraph handles X in a toolbar, not in a separate figure"). This makes understanding the various backends and how they differ much easier for future maintainers.

> What may be a solution to this specific problem could be putting this method and also e.g _fake_click into MNEFigure.

For this particular case it may be that we need to revisit how `_fake_click` and `_fake_keypress` work to allow them to interact with toolbar elements more easily. Or perhaps we will ultimately need higher-level methods that are part of each backend, like `_route_keypress_to_annotation_textbox(key)`, so that the tests simply call that method and the helper methods contain the logic to get the correct window / textbox in focus to receive the key.  But for now, *remember that this PR is supposed to split out matplotlib-specific code from backend-agnostic code*.  You don't need to make the code fully compatible with PyQtGraph yet; if you foresee troublesome areas it is enough to leave code comments saying, e.g., "this method won't translate well to PyQtGraph backend because annotations are handled in a toolbar".  It will be easier to tackle those incommensurabilities one at a time, once this refactoring is done and once the (relatively) straightforward, similar parts of the new backend have been implemented.

Reply:
> Or perhaps we will ultimately need higher-level methods that are part of each backend, like _route_keypress_to_annotation_textbox(key)

If I understand correctly, we would then have a helper-method for each child-element we want to test the interaction with.
Wouldn't the current solution with passing the child-element as an argument be a bit cleaner?

Or we could create a second Base-Class like `FigureBase`, where we would but all abstract methods which would be used by the main browser and its children together (something like `_click_ch_name` for example would only be used by `MNEBrowseFigure` and all the other methods in `BrowserBase` would also be of no use or even interfering with for e.g. `MNEAnnotationFigure`.

Reply:
> Wouldn't the current solution with passing the child-element as an argument be a bit cleaner?

maybe.  I'm not advocating any specific solution to this specific problem right now. The point I'm trying to make is that deciding among the many possible ways to adapt the tests to handle multiple backends will be *easier* if you *wait* to adapt the tests until *after* the conceptual separation of backend-agnostic and MPL-backend-specific code has been done.